api/base/filters.py

```diff
@@ -45,16 +51,24 @@ def __init__(self, *args, **kwargs):
             raise NotImplementedError()
 
     def is_filterable_field(self, key):
+        #  foo -> foo
+        #  foo.bar -> foo
+        #  foo.bar.bas -> foo
+
+        # base_key = key.strip().split('.')[0]
         try:
             return key.strip() in self.serializer_class.filterable_fields
         except AttributeError:
             return key.strip() in self.serializer_class._declared_fields
 
-    # Used so that that queries by _id will work
+
+    # Used so that that queries by _id, jobs, and schools, will work
     def convert_key(self, key):
         key = key.strip()
+        # # If the field is a complex field, then break it apart to rebuild it better/stronger/faster.
         if self.serializer_class._declared_fields[key].source:
-            return self.serializer_class._declared_fields[key].source
+            new_key = self.serializer_class._declared_fields[key].source
```

Comment:
You could probably just assign this to key rather than making a new variable here. Not a big deal.